### PR TITLE
Create an Antora documentation module

### PR DESCRIPTION
**Pull request by Mogztter**
@mojavelinux Should be ready to be consumed by https://github.com/asciidoctor/docs.asciidoctor.org

### CODE REVIEW

**Comment by Mogztter** [comment]
> My only advice would be to consider reusing the asciidoctor.js component. I don't think the CLI really needs to be a separate component, even though it's a separate piece of software. I would simply align it with one of the versions of Asciidoctor.js. That's a flexibility Antora affords you. (For the navigation, you'd need to use a cross-component include).

You mean, the documentation can stay here but I can reuse the same component name and Antora will merge both ? If it's the case that's so cool it didn't know about this feature 🚀

**Comment by mojavelinux** [comment]
You bet! In fact, Couchbase does exactly that for their CLI.

**Comment by Mogztter** [comment]
> For the navigation, you'd need to use a cross-component include.

I need to remove the navigation from this module and add a CLI navigation in the Asciidoctor.js module ?

```
* CLI
** xref:cli-setup/install.adoc[Install guide]
** xref:cli-setup/quick-tour.adoc[A quick tour]
** Integration
*** xref:cli-integration/revealjs.adoc[reveal.js]
```

**Comment by mojavelinux** [comment]
You can use an include in the navigation file for asciidoctor.js.

```
include::cli$partial/nav.adoc[]
```

(where cli is a module)

If it become necessary to nest the navigation, you can use an open block. See https://raw.githubusercontent.com/couchbase/docs-server/release/6.0/modules/ROOT/nav.adoc for examples.

**Comment by Mogztter** [comment]
@mojavelinux I've copied what Couchbase did. I now have a module named `cli` with a subfolder `asciidoctorjs` (the name of the commande) with all the documentation about the Asciidoctor.js CLI. Is that OK ?

**Comment by mojavelinux** [comment]
I don't think the subfolder is needed (unless you anticipate having other CLIs for Asciidoctor.js...but even then).

So I think the path can simply be:

docs/modules/cli/pages/index.adoc

And that would do it.

**Comment by mojavelinux** [comment]
I am aware that having to create a folder for the cli module in the cli repository is redundant. I plan to make the folder configurable so it can still provide the module, but be flat. That's what https://gitlab.com/antora/antora/issues/28 is about.

**Comment by Mogztter** [comment]
> I don't think the subfolder is needed (unless you anticipate having other CLIs for Asciidoctor.js...but even then).

That was the idea since `asciidoctorjs-pdf` is coming :rocket: 

> I'm aware that having to create a folder for the cli module in the cli repository is redundant. I plan to make the folder configurable so it can still provide the module, but be flat. That's what https://gitlab.com/antora/antora/issues/28 is about.

That would be nice for this use case :+1:

**Comment by mojavelinux** [comment]
> That was the idea since asciidoctorjs-pdf is coming rocket

But that I would consider to be a separate component.

**Comment by Mogztter** [comment]
> But that I would consider to be a separate component.

Ok then your proposal makes sense `docs/modules/cli/pages/index.adoc` :+1:

**Comment by mojavelinux** [comment]
\o/ I'm excited to see this pattern in use within the Asciidoctor docs.

File: docs/modules/cli/pages/index.adoc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+= Asciidoctor.js CLI Documentation
+
+The Command Line Interface (CLI) for xref:master@asciidoctor.js:ROOT/index.adoc[Asciidoctor.js].
```
Comment (mojavelinux) [inline docs/modules/cli/pages/index.adoc]: You don't need to include the version and module in this case as these are the default values. So you can shorten this to:

```
xref:asciidoctor.js::index.adoc[Asciidoctor.js]
```

But since you are fusing the CLI docs with the processor docs, then you just need the module reference (which, for now, you can't drop even though it's ROOT):

```
xref:ROOT:index.adoc[Asciidoctor.js]
```

File: docs/modules/cli/pages/install.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+If the module installed successfully, its command line interface (CLI) will be available on your `PATH`.
+To confirm that Asciidoctor.js CLI is available, execute:
+
+[source]
+$ asciidoctorjs --version
+
+You should see information about the Asciidoctor.js CLI version printed in the terminal.
+
+[source]
+1.5.6-rc.1
```
Comment (mojavelinux) [inline docs/modules/cli/pages/install.adoc]: Isn't this now 1.5.9?

File: docs/modules/cli/pages/integration-revealjs.adoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+In this section, you'll learn:
+
+* [x] How to install the required dependencies.
+* [x]  What attributes you must define.
+* [x] How to convert an AsciiDoc file to a _reveal.js_ presentation.
+
+== Prerequisites
+
+We assume that you already have a fully working version of Asciidoctor.js CLI installed and available in your `PATH`.
+If this is not the case, please read the xref:setup/install.adoc[installation instructions].
```
Comment (mojavelinux) [inline docs/modules/cli/pages/integration-revealjs.adoc]: I think you have an extra path segment in this xref. I don't see a "setup" folder.

File: docs/modules/cli/pages/integration-revealjs.adoc [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+To convert an AsciiDoc file to a _reveal.js_ presentation, we will need to install the following module:
+
+[source]
+$ npm i asciidoctor-reveal.js
+
+NOTE: If you prefer Yarn over npm, you can install Asciidoctor.js reveal.js using `yarn add asciidoctor-reveal.js`
```
Comment (mojavelinux) [inline docs/modules/cli/pages/integration-revealjs.adoc]: Missing a period at the end of this sentence.

File: docs/modules/cli/pages/install.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+== Install Asciidoctor.js CLI
+
+You must have Asciidoctor.js installed to use it.
+
+Install Asciidoctor.js and the CLI globally and you'll have access to the `asciidoctorjs` command anywhere on your system.
+Open a terminal and type:
+
+[source]
+$ npm i -g asciidoctor.js asciidoctor-cli
```
Comment (mojavelinux) [inline docs/modules/cli/pages/install.adoc]: Generally I prefer commands to use the following syntax (offset by one space):

```
 $ npm i -g asciidoctor.js asciidoctor-cli
```

Then we can use an extension to beauty them up.

File: docs/modules/cli/pages/integration-revealjs.adoc [hunk 6]
```diff
@@ -0,0 +1,56 @@
+= Integrate with reveal.js
+
+In this section, you'll learn:
+
+* [x] How to install the required dependencies.
+* [x]  What attributes you must define.
```
Comment (mojavelinux) [inline docs/modules/cli/pages/integration-revealjs.adoc]: There's an extra space in front of the text of this list item.

File: docs/modules/cli/pages/integration-revealjs.adoc [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+Please remember that if you want this converter to be available anywhere on your system,
+you will have to install this module with the `--global` option with `npm` or the `global` command with Yarn.
+
+== Convert an AsciiDoc presentation into slides
+
+Once this module is installed, we create a file named _presentation.adoc_ with the following content:
+
+.presentation.adoc
+```adoc
```
Comment (mojavelinux) [inline docs/modules/cli/pages/integration-revealjs.adoc]: I think we should use a consistent authoring style for code blocks. Let's use the formal syntax:

```
[source,asciidoc]
----
----
```

File: docs/modules/cli/pages/integration-revealjs.adoc [hunk 8]
```diff
@@ -0,0 +1,56 @@
+= Integrate with reveal.js
+
+In this section, you'll learn:
+
+* [x] How to install the required dependencies.
+* [x]  What attributes you must define.
+* [x] How to convert an AsciiDoc file to a _reveal.js_ presentation.
```
Comment (mojavelinux) [inline docs/modules/cli/pages/integration-revealjs.adoc]: I don't think reveal.js should be italicized here.

Comment (mojavelinux) [reply docs/modules/cli/pages/integration-revealjs.adoc]: You could make it a link though.

File: docs/modules/cli/pages/integration-revealjs.adoc [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+$ npm i asciidoctor-reveal.js
+
+NOTE: If you prefer Yarn over npm, you can install Asciidoctor.js reveal.js using `yarn add asciidoctor-reveal.js`
+
+Please remember that if you want this converter to be available anywhere on your system,
+you will have to install this module with the `--global` option with `npm` or the `global` command with Yarn.
+
+== Convert an AsciiDoc presentation into slides
+
+Once this module is installed, we create a file named _presentation.adoc_ with the following content:
```
Comment (mojavelinux) [inline docs/modules/cli/pages/integration-revealjs.adoc]: When referring to paths, we should always add the path role.

```
[.path]_presentation.adoc_
```

That way we can style them consistently.

File: docs/modules/cli/pages/quick-tour.adoc [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+== Invoke Asciidoctor.js from the CLI
+
+To invoke Asciidoctor.js from the CLI and convert an `.adoc` file, execute:
+
+[source]
+$ asciidoctorjs document.adoc
+
+This will use the built-in defaults for options and create a new file in the same directory as the input file, with the same base name, but with the `.html` extension.
+
+The xref:options-and-parameters.adoc[Command Line Options and Parameters] describes the available Asciidoctor CLI options and parameters.
```
Comment (mojavelinux) [inline docs/modules/cli/pages/quick-tour.adoc]: I had to look this up recently and I found that when talking about a command line tool, you only need to say "options". Parameters is unnecessary here.

**Comment by Mogztter** [comment]
Wow thanks for the thorough review! I will edit my changes tomorrow 👍

File: docs/modules/cli/pages/install.adoc (hunk 2 above)
Comment (mojavelinux) [reply docs/modules/cli/pages/install.adoc]: :+1:

File: docs/modules/cli/pages/index.adoc (hunk 1 above)
Comment (mojavelinux) [reply docs/modules/cli/pages/index.adoc]: Yep. I haven't figured out how to express it without ending up with a block-like macro (xref::index.adoc[Asciidoctor.js])

**Comment by Mogztter** [comment]
@mojavelinux I think I've fixed all your comments, can we proceed ?
